editor/store/selectors.js

```diff
@@ -1397,6 +1397,7 @@ export const getInserterItems = createSelector(
 				isDisabled,
 				utility: calculateUtility( blockType.category, count, isContextual ),
 				frecency: calculateFrecency( time, count ),
+				hasChildBlocks: hasChildBlocks( blockType.name ),
```

Comment:
How is the cache bust if we proceed to register a block defining `blockType.name` as a valid `parent` ?

Reply:
We clean the cache each time state.blockTypes changes, so when we register a block the cache is invalidated.

Reply:
>We clean the cache each time state.blockTypes changes, so when we register a block the cache is invalidated.

But `state.blockTypes` (from `core/blocks` store) is not considered when we call `getInserterItems` (from `core/editor` store).

Reply:
I understand now the problem, getInserterItems is cached and depends on core/blocks store when core/blocks change the cache may not be updated. I think this is an already existing problem. Before if we register a new block a new item should exist but the cache may not be updated, it is not easy to spot because the probably the cache updates a lot.

Thank you for clarifying this issue I will try to propose a solution to this problem!